client/src/stores/toolStore.ts

```diff
@@ -75,12 +75,19 @@ export interface PanelView {
 export const useToolStore = defineStore("toolStore", () => {
     const currentPanelView: Ref<string> = useUserLocalStorage("tool-store-view", "");
     const defaultPanelView: Ref<string> = ref("");
-    const toolsById = ref<Record<string, Tool>>({});
+    const toolsById = shallowRef<Record<string, Tool>>({});
     const toolResults = ref<Record<string, string[]>>({});
     const panel = ref<Record<string, Record<string, Tool | ToolSection>>>({});
     const panelViews = ref<Record<string, PanelView>>({});
     const loading = ref(false);
 
+    const searchWorker = ref<Worker | undefined>(undefined);
+    // TODO: Where do we terminate the worker?
+    // onScopeDispose(() => {
+    //     searchWorker.value?.terminate();
+    //     searchWorker.value = undefined;
```

Comment:
How important is it to terminate the worker ? It does seem like the callback holds on to `currentFavorites.value.tools` (which makes we wonder, does the reactivity there work cross thread?), but if we only ever create a single worker that seems fine ? Do you know if the callback keeps a reference to the ToolSearch Component ?

Maybe you could remove just the callback when you unmount the tool search ?

Reply:
We definitely only ever create the one worker here, because we only create it if it is undefined in the store.

I'm trying to investigate where you're getting the value for the "callback" though?

Reply:
So, upon investigation, I can see that the onmessage handler retains its value outside the scope of the `ToolSearch` component. Does it keep a "reference" to the refs inside of it outside the component, I can't seem to be able to inspect that but to be safe: I'll do what you recommended here and just remove the callback `onUnmounted` since there is no harm in that.